appscale/tools/agents/azure_agent.py

```diff
@@ -206,11 +206,18 @@ def configure_instance_security(self, parameters):
       AgentRuntimeException: If security features could not be successfully
         configured in the underlying cloud.
     """
+    is_autoscale = parameters['autoscale_agent']
+
+    # While creating instances during autoscaling, we do not need to create a
+    # new keypair or a resource group. We just make use of the existing one.
+    if is_autoscale in ['True', True]:
```

Comment:
No, this gets set as a boolean in the Infrastructure manager client, which then gets passed around to the agent as an uppercase 'True' as a string. 

I ran into this issue when I was testing autoscaling using the tools agents, the parameters are passed around the server side as a string, but on the tools side they are passed as a boolean. Hence the check for both. Its passed from the Infrastructure Manager client to the manager via a SOAP call, so I believe that can be done only as a string.